canu/generate/switch/config/config.py

```diff
@@ -519,7 +588,7 @@ def get_pair_connections(nodes, switch_name):
     return connections
 
 
-def get_switch_nodes(switch_name, shcd_node_list, factory):
+def get_switch_nodes(switch_name, shcd_node_list, factory, sls_variables):
```

Comment:
This is needed because we need to map the cabinets/rack from the shcd_node_list to the sls_variables file in order to apply the VLAN to the port. 

I agree that this should be in different location/seperate function, however I think a bigger refactor would need to be done to fix other issues that come with this.